src/pages/overlay/hooks/useSettings.tsx

```diff
@@ -19,23 +19,45 @@ const settings = {
     type: "boolean",
     process: (value: any) => !!value,
     devOnly: false,
+    configurable: true,
   },
   disableOverlayHiding: {
     title: "(DEV) Prevent app hiding automatically",
     type: "boolean",
     process: (value: any) => !!value,
     devOnly: true,
+    configurable: true,
   },
+  openedMenu: {
+    title: "Menu that was last opened",
+    type: "string",
+    process: (value: any) =>
+      typeof value === "string" ? value : "ambassadors",
+    devOnly: false,
+    configurable: false,
+  },
+};
+
+type SettingsProcessReturnTypes = {
+  boolean: boolean;
+  string: string;
 };
 
 type SettingsKey = keyof typeof settings;
 
 type StoredSettings = {
-  [key in SettingsKey]: ReturnType<(typeof settings)[key]["process"]>;
+  [key in SettingsKey]: key extends keyof SettingsProcessReturnTypes
```

Comment:
This doesn't seem right? `keyof SettingsProcessReturnTypes` is going to be `boolean | string`, but `key in SettingsKey` is going to be `"disableChatPopup" | "disableOverlayHiding" | "openedMenu"`?

Reply:
Sorry, I'm realizing now the typing changes I made are nonsense and unnecessary. Pulled most of it out, should be saner now.